### DISCUSSION TITLE
edit page wont load content

### DISCUSSION THREAD

**Discussion by d-opri** [neuefische/questions#520] [Web] [answered]
moin moin

i am working to implement edit functionality. the link from the details page leads to the edit page, but wont load the content of the bill. my question would be how to pass the data to my form component. in the component i save all the inputs in seperate usestates, do i need to compile them into one so i can pass them as one to my my form component on the edit page? or what else am i missing?

[my pr](https://github.com/d-opri/Kroko-Capstone/pull/12)

thankful for any advice

**Comment by mbosselmann** [comment, marked as answer]
Hi Danie,
yes you need to pass the bill data as prop to your `BillForm` component. Currently you do not use this prop in your `BillForm` component. Add it like this:

```js
export default function BillForm({ addBill, bill }) {
...
}
```

And then you can update your controlled component and depend on whether the prop is defined or not which value should be displayed. An example for your input element for your title could look like this:

```js
          <input
            type="text"
            onChange={(event) => setTitle(event.target.value)}
            value={bill ? bill.title : title}
            name="title"
            id="title"
            pattern="/^[a-zA-Z0-9 ]{2,30}$/"
            maxLength={20}
            minLength={2}
          />
```

Try this out. I hope this does help. :)

**Reply by d-opri** [reply]
my saviour !!  it worked and shows my values now. however i cannot edit the input fields... have i forgotten something obvious? i thought i needed to create a new bill state and change all my current on change events to update that state but then i cant pass bill as a prop anymore since it will be declared already. or is my edit page incorrect?

```

export default function EditBillPage() {
  const router = useRouter();
  const { id } = router.query;
  const { isReady } = router;

  const { data: bill, isLoading, error } = useSWR(`/api/bills/${id}`);

  async function editBill(bill) {
    await fetch(`/api/bills/${id}`, {
      method: "PATCH",
      body: JSON.stringify(bill),
    });
    router.push("/");
  }
  if (!isReady || isLoading || error) return <h2>Loading Bill..</h2>;

  return (
    <>
      <StyledLink href={`/bills/${id}`}>X</StyledLink>
      <Title>Edit Bill</Title>

      <BillForm addBill={editBill} bill={bill} />
    </>
  );
}
```

i fetch the data, then i send my patch request and get my data back which i pass to the component. i cant think of anything other than my on change being the cause.

thanks again

**Reply by d-opri** [reply]
actually i figured it out on my own! changed it so now the bill prop is in the useStates and not input elements :)  thanks again for the great help :)
